crates/biome_js_analyze/src/lint/nursery/no_restricted_imports.rs

```diff
@@ -60,42 +286,705 @@ declare_lint_rule! {
+                current = current.parent()?;
+            }
+        } else if current.kind() == JsSyntaxKind::JS_STATIC_MEMBER_EXPRESSION {
+            // #2: **(import("")).then**(...)
+            let static_member_expr = current.cast::<JsStaticMemberExpression>()?;
+            let member_name = static_member_expr.member().ok()?;
+            if member_name.as_js_name()?.text() != "then" {
+                return None;
+            }
+            current = static_member_expr.syntax().parent()?;
+
+            if current.kind() == JsSyntaxKind::JS_CALL_EXPRESSION {
+                // #2: **(import("")).then(...)**
+                let then_call_expr = current.cast::<JsCallExpression>()?;
+                let then_call_arg = then_call_expr
+                    .arguments()
+                    .ok()?
+                    .args()
+                    .into_iter()
+                    .next()?
+                    .ok()?
+                    .as_any_js_expression()?
+                    .clone()
+                    .omit_parentheses();
+
+                return match then_call_arg {
+                    // then(... => ...)
+                    AnyJsExpression::JsArrowFunctionExpression(arrow_expr) => {
+                        match arrow_expr.parameters().ok()? {
+                            // then(arg => ...)
+                            AnyJsArrowFunctionParameters::AnyJsBinding(binding) => {
+                                Some(AnyJsBindingPattern::AnyJsBinding(binding))
+                            }
+                            // then ({ ... } => ...)
+                            AnyJsArrowFunctionParameters::JsParameters(parameters) => Some(
+                                parameters
+                                    .items()
+                                    .into_iter()
+                                    .next()?
+                                    .ok()?
+                                    .as_any_js_formal_parameter()?
+                                    .as_js_formal_parameter()?
+                                    .binding()
+                                    .ok()?,
+                            ),
+                        }
+                    }
+                    // then(function(...) { ... })
+                    AnyJsExpression::JsFunctionExpression(function_expr) => Some(
+                        function_expr
+                            .parameters()
+                            .ok()?
+                            .items()
+                            .into_iter()
+                            .next()?
+                            .ok()?
+                            .as_any_js_formal_parameter()?
+                            .as_js_formal_parameter()?
+                            .binding()
+                            .ok()?,
+                    ),
+                    _ => None,
+                };
+            }
+        }
+
+        // #1: const { ... } = **(await (import("")))**
```

Comment:
Why do you use `#1` and `#2`? Just curious, because I know what they mean

Reply:
I tried different variations (`// (Variant 1)`, `// (1)`, `// 1:`, ...) and found `#1` to be the best compromise between:
- Not too long (which would distract from quickly spotting the interesting part of the comment)
- Visually "pops out" just enough to be recognizable while scanning over the source code

It was mainly a tool for me while writing the code, but I left it so it might help the next person visiting the code. Feel free to change if you feel inclined :)